### ISSUE TITLE
event.res can be a negative number inside an unsigned type

### ISSUE THREAD

**Issue by ali1234** [vpelletier/python-libaio#1]
If you send a read or write event to functionfs and the USB link disconnects (other side closes it or whatever), then the read() or write() call will return -108,

ESHUTDOWN 108 Cannot send after transport endpoint shutdown

Which is totally reasonable and everything. But the io_event struct says e.res is a weird, padded unsigned type, so python interprets it as a really big positive number. I can't for the life of me figure out how to cast it to signed nicely, in order to catch these errors.

**Comment by vpelletier** [comment]
Hello,

I also had this issue when I started working on the bidirectional chat example, and I changed these fields to become signed in 09384b5de3cf51a233a10b5d67e5ca6f24a245fd , which I then released as 0.2 .

I have no idea why libaio did all these explicit padding things. Kernel structs do not, and C already does that padding when not told to pack. but I do not trust myself to accurately predict memory layout of structures on more than one arch at a time, so I just sticked to what I found in libaio header.
